dll/win32/uxtheme/errinfo.c

```diff
@@ -0,0 +1,174 @@
+
+static HRESULT
+UXTHEME_FormatParseMessage(
+    _In_ PTMERRINFO pErrInfo,
+    _Out_ LPWSTR pszDest,
+    _In_ SIZE_T cchDest)
+{
+    DWORD nID;
+    HMODULE hMod, hUxTheme;
+    WCHAR szFullPath[_MAX_PATH];
+    WCHAR szDrive[_MAX_DRIVE + 1], szDir[_MAX_DIR], szFileName[_MAX_FNAME], szExt[_MAX_EXT];
+    BOOL ret;
+
+    nID = LOWORD(pErrInfo->nID);
+    if (!GetModuleFileNameW(NULL, szFullPath, _countof(szFullPath)))
+        return S_OK;
+
+    _wsplitpath(szFullPath, szDrive, szDir, szFileName, szExt);
+    if (lstrcmpiW(szFileName, L"packthem") == 0)
+    {
+        hMod = GetModuleHandleW(NULL);
+        if (UXTHEME_FormatLocalMsg(hMod, nID, pszDest, cchDest, szDrive, pErrInfo))
+            return S_OK;
+    }
+
+    hUxTheme = LoadLibraryW(L"uxtheme.dll");
+    if (!hUxTheme)
+        return E_FAIL;
+    ret = UXTHEME_FormatLocalMsg(hUxTheme, nID, pszDest, cchDest, szDrive, pErrInfo);
+    FreeLibrary(hUxTheme);
+
+    return ret ? S_OK : UXTHEME_MakeErrorLast();
+}
+
+HRESULT
+UXTHEME_GetThemeParseErrorInfo(_Inout_ PPARSE_ERROR_INFO pInfo)
+{
+    PTMERRINFO pErrInfo;
+    HRESULT hr;
+
+    if (pInfo->cbSize != sizeof(PARSE_ERROR_INFO))
+        return UXTHEME_MakeError32(E_INVALIDARG);
+
+    pErrInfo = UXTHEME_GetParseErrorInfo(TRUE);
+    if (!pErrInfo)
+        return UXTHEME_MakeError32(E_OUTOFMEMORY);
+
+    hr = UXTHEME_FormatParseMessage(pErrInfo, pInfo->ErrInfo.szPath0,
+                                    _countof(pInfo->ErrInfo.szPath0) +
+                                    _countof(pInfo->ErrInfo.szPath1));
+    if (FAILED(hr))
+        return hr;
+
+    pInfo->ErrInfo.nID = pErrInfo->nID;
+    pInfo->ErrInfo.dwError = pErrInfo->dwError;
+    StringCchCopyW(pInfo->ErrInfo.szPath2, _countof(pInfo->ErrInfo.szPath2), pErrInfo->szPath2);
+    StringCchCopyW(pInfo->ErrInfo.szPath3, _countof(pInfo->ErrInfo.szPath3), pErrInfo->szPath3);
+    return hr;
+}
+
+// Parser should use this function on failure
+HRESULT
+UXTHEME_MakeParseError(
+    _In_ UINT nID,
+    _In_ LPCWSTR pszPath0,
+    _In_ LPCWSTR pszPath1,
+    _In_ LPCWSTR pszPath2,
+    _In_ LPCWSTR pszPath3,
+    _In_ DWORD dwError)
+{
+    PTMERRINFO pErrInfo = UXTHEME_GetParseErrorInfo(TRUE);
+    if (pErrInfo)
+    {
+        pErrInfo->nID = nID;
+        pErrInfo->dwError = dwError;
+        StringCchCopyW(pErrInfo->szPath0, _countof(pErrInfo->szPath0), pszPath0);
+        StringCchCopyW(pErrInfo->szPath1, _countof(pErrInfo->szPath1), pszPath1);
+        StringCchCopyW(pErrInfo->szPath2, _countof(pErrInfo->szPath2), pszPath2);
+        StringCchCopyW(pErrInfo->szPath3, _countof(pErrInfo->szPath3), pszPath3);
+    }
+    return HRESULT_FROM_WIN32(ERROR_UNKNOWN_PROPERTY);
+}
+
+/*************************************************************************
+ *  GetThemeParseErrorInfo (UXTHEME.48)
+ */
+HRESULT WINAPI
+GetThemeParseErrorInfo(_Inout_ PPARSE_ERROR_INFO pInfo)
+{
+    if (IsBadWritePtr(pInfo, sizeof(*pInfo)))
+        return UXTHEME_MakeError32(E_POINTER);
```

Comment:
```suggestion
        return E_POINTER;
```
it's already an hresult error.